src/app/dynamic-forms/cinchy-dynamic-forms.component.ts

```diff
@@ -420,41 +429,44 @@ export class CinchyDynamicFormsComponent implements OnInit, OnChanges {
       this.form = await this._formHelperService.generateForm(this.formMetadata, this.rowId,tableEntitlements);
       this._formHelperService.fillWithSections(this.form, this.formSectionsMetadata);
 
-      this.cinchyQueryService.getFormFieldsMetadata(this.formId).subscribe(
+      this._formFieldMetadataSubscription = this.cinchyQueryService.getFormFieldsMetadata(this.formId).subscribe(
+        
         async (formFieldsMetadata) => {
+            let selectedLookupRecord;
+            
+            await this._formHelperService.fillWithFields(this.form, this.rowId, this.formMetadata, formFieldsMetadata, selectedLookupRecord, tableEntitlements);
+            if (this.lookupRecordsList) {
+              selectedLookupRecord = this.lookupRecordsList.find((record: ILookupRecord) => {
 
-          let selectedLookupRecord = this.lookupRecordsList.find((record: ILookupRecord) => {
-
-            return (record.id === this.rowId);
-          });
-
-          await this._formHelperService.fillWithFields(this.form, this.rowId, this.formMetadata, formFieldsMetadata, selectedLookupRecord, tableEntitlements);
-
-          // This may occur if the rowId is not provided in the queryParams, but one is
-          if (this.rowId !== null) {
-            if (!selectedLookupRecord) {
-              this.appStateService.setRecordSelected(null);
+              return (record.id === this.rowId);
+            });
+          }
+            // This may occur if the rowId is not provided in the queryParams, but one is
```

Comment:
Looks like I forgot to finish this comment. Should read:

`// This may occur if the rowId is not provided in the queryParams, but one is available in localstorage from a previous instance`
`// of the app. If the given rowId doesn't match a record on this table, clear it`